src/nemo_evaluator/solvers/harbor.py

```
@@ -1160,9 +1239,39 @@ async def _wait_for_agent(
+                        "    except ValueError:\n"
+                        "        continue\n"
+                        "    comm, args = parts[1], parts[2]\n"
+                        "    if pid != me and 'python' in comm and '/installed-agent/run_agent.py' in args:\n"
+                        "        os.kill(pid, signal.SIGTERM)\n"
+                        "PY",
+                        timeout_sec=10,
+                    )
+                    done_after_signal, _ = await asyncio.wait({agent_task}, timeout=30.0)
+                    if agent_task in done_after_signal:
+                        logger.info("HarborSolver: timed-out agent flushed after SIGTERM")
+                except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)